### DISCUSSION TITLE
Cockpit as a replacement for OpenLMI

### DISCUSSION THREAD

**Discussion by rousku** [cockpit-project/cockpit#17127] [Q&A] [answered]
Hi!
I'm looking for an API implementation that I could use to manage some servers. The API should provide means to start/stop services, configure networks, etc. and the API should be extendable so that I could implement custom server management tasks. By reading the documentation of the OpenLMI, it looks like it was created for these use cases. However, www.openlmi.org is down, there are no packages for Ubuntu, RHEL8 dropped OpenLMI related packages, so it seems the project is dead.

Officially Cockpit seems to support only web interface but would it make sense to utilize the WebSocket interface for programmatic management of servers?

**Comment by martinpitt** [comment, marked as answer]
I don't think that would make too much sense. The whole cockpit-ws/websocket/cockpit-bridge functionality exists solely for web browsers and thus the interactive UI that Cockpit provides. The bridge protocol allows you to do do D-Bus calls, file access/writing, etc., all of which just wrap existing Linux APIs.

If you want to manage servers programmatically, then just SSH is the right thing to do. Especially with [Ansible](https://www.ansible.com/), which is meant for programmatically managing servers, and do these things. For example, the [service module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/service_module.html) can start/stop/enable/disable services. Moreover, Ansible is written in a configuration management language (specify target state and be idempotent) instead of in an operational language that you need for UIs.
